From 00ac09cb74a6042435af72fd9153b1268939ea51 Mon Sep 17 00:00:00 2001
From: Zhao Yakui <yakui.zhao@intel.com>
Date: Wed, 21 Aug 2019 11:43:50 +0800
Subject: [PATCH 1434/1494] Revert "Reduce the unnecessary interrupt injection
 for small time interval"

This reverts commit bbb891728d82834ec450f6a61792f715f4ec3013.

---
 drivers/gpu/drm/i915/gvt/gvt.h  |  1 -
 drivers/gpu/drm/i915/gvt/mpt.h  | 14 --------------
 drivers/gpu/drm/i915/gvt/vgpu.c |  1 -
 3 files changed, 16 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/gvt.h b/drivers/gpu/drm/i915/gvt/gvt.h
index b0073ed..cf68915 100644
--- a/drivers/gpu/drm/i915/gvt/gvt.h
+++ b/drivers/gpu/drm/i915/gvt/gvt.h
@@ -240,7 +240,6 @@ struct intel_vgpu {
 	unsigned long long *cached_guest_entry;
 	bool ge_cache_enable;
 	bool entire_nonctxmmio_checked;
-	ktime_t vgpu_msi_time;
 };
 
 /* validating GM healthy status*/
diff --git a/drivers/gpu/drm/i915/gvt/mpt.h b/drivers/gpu/drm/i915/gvt/mpt.h
index 6030e41..14fa2ea 100644
--- a/drivers/gpu/drm/i915/gvt/mpt.h
+++ b/drivers/gpu/drm/i915/gvt/mpt.h
@@ -129,21 +129,7 @@ static inline int intel_gvt_hypervisor_inject_msi(struct intel_vgpu *vgpu)
 	u16 control, data;
 	u32 addr;
 	int ret;
-	ktime_t time_current;
 
-	time_current = ktime_get_raw();
-
-	if (ktime_before(time_current,
-			 ktime_add_ns(vgpu->vgpu_msi_time, 10000))) {
-		/*
-		 * vgpu->vpgu_msi_time records the time of last interrupt
-		 * injection. when the time interval is too small,
-		 * the interrupt injection will be skipped.
-		 */
-		return 0;
-	}
-
-	vgpu->vgpu_msi_time = time_current;
 	control = *(u16 *)(vgpu_cfg_space(vgpu) + MSI_CAP_CONTROL(offset));
 	addr = *(u32 *)(vgpu_cfg_space(vgpu) + MSI_CAP_ADDRESS(offset));
 	data = *(u16 *)(vgpu_cfg_space(vgpu) + MSI_CAP_DATA(offset));
diff --git a/drivers/gpu/drm/i915/gvt/vgpu.c b/drivers/gpu/drm/i915/gvt/vgpu.c
index 17221f9..5411b40 100644
--- a/drivers/gpu/drm/i915/gvt/vgpu.c
+++ b/drivers/gpu/drm/i915/gvt/vgpu.c
@@ -505,7 +505,6 @@ struct intel_vgpu *intel_gvt_create_vgpu(struct intel_gvt *gvt,
 		/* calculate left instance change for types */
 		intel_gvt_update_vgpu_types(gvt);
 	mutex_unlock(&gvt->lock);
-	vgpu->vgpu_msi_time = ktime_get_raw();
 
 	return vgpu;
 }
-- 
2.7.4

